### DISCUSSION TITLE
Connection R5

### DISCUSSION THREAD

**Discussion by Like-electronics** [jwillemsen/saj_collector#25] [General]
Sinds kort hebben we een installatie van Zonneplan met daarbij een SAJ R5 omvormer. De gegevens worden naar de cloud verstuurd door een 4G modem. Er is dus geen netwerkaansluiting of wifi. Hoe kan ik nu verbinding maken met de omvormer om de gegevens lokaal of via PVoutput binnen te krijgen?

**Comment by jwillemsen** [comment]
Ik heb zelf geen R5 dus kan dit lastig beantwoorden, de SAJ die ik heb heeft een ethernet aansluiting en is op die manier makkelijk uit te lezen.

Als ik de [manual](https://solarnrg.nl/downloads/handleidingen/solarnrg-handleiding-saj-wifi-module.pdf) van de SAJ wifi module lees dan is er een optie om de omvormer via wifi te verbinden, dan zou hij wel via het lokale netwerk te bereiken zijn.

Van de 4G module kan ik weinig [documentatie](https://nl.saj-electric.com/product/info-645626e0aab3445284d3c81556fdd59f.html) vinden, er staat wel dat die een integrated webpage heeft, misschien dat die ook de data geeft?

**Comment by Like-electronics** [comment]
Er is wel een RS485 aansluiting.

Op vr 12 feb. 2021 15:28 schreef Johnny Willemsen <notifications@github.com

**Reply by jwillemsen** [reply]
De RS485 uitlezen is een heel project op zich, zie https://www.solartoday.nl/wp-content/uploads/saj-plus-series-inverter-modbus-protocal.pdf

**Reply by jwillemsen** [reply]
Op zich zou dit te doen moeten zijn, RS485/modbus support is er ook voor Ruby, ik heb echter zelf geen R5 dus ik kan niets testen wat dit een trial/error effort zou zijn en daar heb ik op dit moment niet echt tijd voor. De ontwikkeling van deze Ruby gem viel op zich mee omdat ik zelf een SAJ heb hangen

**Comment by Like-electronics** [comment]
Heb je meer gegevens van dat modem en webpage?


Op vr 12 feb. 2021 15:28 schreef Johnny Willemsen <notifications@github.com

**Comment by Like-electronics** [comment]
Het 4G modem heeft wel bluetooth.
Als je hier verbinding mee wilt maken, krijg je een melding dat dat alleen
gaat met een app. De app van eSolar geïnstalleerd en een account aangemaakt
Serienummer en gegevens ingevoerd. Allemaal prima en werkend gekregen. Er
komen echter geen gegevens binnen.
Er is nl geen account bij SAJ.
Zonneplan heeft een koppeling met een Europese cloud vanwege bescherming
privacy. Wie of waar dat is.......

Op vr 12 feb. 2021 15:31 schreef F f <fg.wouters@gmail.com>:

**Reply by jwillemsen** [reply]
Geen idee waar dat is. Het zou technisch mogelijk moeten zijn om de data weer uit het SAJ portal te halen, maar dat is heel wat uitzoek werk. Het zou veel eenvoudiger en handiger zijn als je in het SAJ portal de pvoutput API key kunt invullen en dat zij het gelijk doorsturen naar pvoutput. Heb ik SAJ al een paar keer op gewezen, maar nooit echt een antwoord op gehad

**Comment by jwillemsen** [comment]
Er is blijkbaar een manier om de data uit de SAJ portal te halen. Via SAJ support heb  ik 3 keys gekregen die ik daarvoor nodig heb, maar ik heb nog geen manual van de API zelf, daar heb ik ze in het weekend om gevraagd. is er bij zonneplan dan wel een portal beschikbaar waar je het gebruik in kunt zien?

**Comment by Like-electronics** [comment]
Hallo Johnny,

Bij Zonneplan hebben ze een eigen portal in Europa vanwege privacy.
Op mijn omvormer zit een 4g stick die contact maakt met de cloud.
Het is dus niet mogelijk met wifi te werken. Wel zit er een RS-485
(modbus) aansluiting op.
Daarvan is het protocol wel bekend.
Deze kan ik koppelen met Domoticz maar heb geen programma hiervoor.
Nu maak je verbinding met een app van zonneplan zelf. Zie screen shot.
Op de 4g stick zit ook nog bluetooth.
Deze staat aan maar weet niet welke app hiervoor nodig is. Heb de app van
SAJ geprobeerd maar deze werkt niet. Waarschijnlijk  omdat deze omvormer
niet op het SAJ-portal zit.



Op wo 24 feb. 2021 10:18 schreef Johnny Willemsen <notifications@github.com

**Reply by jwillemsen** [reply]
Dan lijkt me de RS-485 de beste koppeling, ik heb echter geen R5 om dit te uit te zoeken. Hopelijk kun je ergens op internet iemand vinden die dat heeft uitgezocht.

**Comment by Like-electronics** [comment]
Gisteren nog contact gehad met zonneplan over het uitlezen van de omvormer.
Hierop de volgende reactie gehad:
Goedemorgen, de Zonneplan ONE omvormer is inderdaad voorbereid op het
aanmaken van een bluetooth verbinding of API key. We zijn bezig met deze
ontwikkelingen, maar het is nog niet klaar voor gebruik. Je kunt de
opbrengsten nu alleen uitlezen via de Zonneplan app. Zodra er nieuwe
features beschikbaar zijn, dan delen we dat via de app

Op wo 24 feb. 2021 15:46 schreef Johnny Willemsen <notifications@github.com

**Reply by jwillemsen** [reply]
Hmm, waarom leveren ze dan bij Zonneplan dan niet de support om via hun te uploaden naar pvoutput, dat zou het echt een stuk makkelijker maken. Invullen van de pvoutput API key in je profile en dan kunnen zij het als batch doorzetten naar pvoutput.

**Reply by jwillemsen** [reply]
Mooiste zou zijn als de SAJ/Zonneplan bij pvoutput via de auto uploader zou werken (https://pvoutput.org/help/auto_uploader.html?highlight=solaredge#getting-started), maar dat zou redelijk wat werk zijn.
